README.md

```diff
@@ -10,13 +10,62 @@
   <image alt="License" src="https://img.shields.io/github/license/castle/castle-cloudflare-worker-demo"/>
 </div>
 
+## Overview
+
+The Castle Cloudflare Worker allows you to put Castle's risk engine right on the edge, in a Cloudflare worker.
+
+## How it works
+Once you've installed the worker with the default settings, the worker will listen for POSTs to the 
+
+`/users/sign_up`
+
+route.
+
+The POST must include a Castle client_id. Click here to learn more about how to include a Castle client_id in a POST.
+
+When the worker receives the POST, it will in turn make a POST to Castle, and receive a risk score in return. If the risk score is above the `riskThreshold`, then the worker will respond with a 403. If the risk score is below the `riskThreshold`, the worker will respond with a 200.
+
+(In production, you would simply forward the request to the upstream service rather than respond with a 200.)
+
+A simple way to test the service is to change the `riskThreshold` to 0, and then make a request from a different browser.
+
+## Prerequisites
+
+You'll need a Castle account and a Cloudflare account to get started.
+
+### Castle
+
+If you don't have a Castle account already, you can [set up a free trial](https://dashboard.castle.io/signup/new). You will need your Castle API Secret, which can be found in the Settings section of your Castle dashboard.
+
+### Cloudflare
+
+If you're going to use the `Deploy with Workers` option (see below), you'll need your Cloudflare account ID and an API Token. (This integration has been tested with a Cloudflare API Token created with the Cloudflare "Edit Worker" token template.)
```

Comment:
I'd remove `(This integration has been tested with a Cloudflare API Token created with the Cloudflare "Edit Worker" token template.)` section or rephrase it. What the end-user needs to know, is what kind of permissions does this token require ("Edit Worker"), but not how the integration has been tested

Reply:
I'll remove it.